test/jdk/java/awt/Frame/InitialMaximizedTest/InitialMaximizedTest.java

```diff
@@ -21,197 +21,82 @@
-  @author Valeriy Ushakov: area=toplevel
-  @run applet/manual=yesno InitialMaximizedTest.html
+ * @test
+ * @bug 4464714 6365898
+ * @key headful
+ * @summary Frames cannot be shown initially maximized
 */
 
-
-/**
- * InitialMaximizedTest.java
- *
- * summary:
- */
-
-import java.applet.Applet;
-import java.awt.*;
-
-
-public class InitialMaximizedTest extends Applet
-{
-    Frame f;
-
-    public void init()
-    {
-        this.setLayout (new BorderLayout ());
-
-        String[] instructions =
-        {
-            "This test creates a frame that is initially maximized.",
-            "Press PASS if frame is shown initially maximized, else press FAIL"
-        };
-        Sysout.createDialogWithInstructions( instructions );
-
-    }//End  init()
-
-    public void start ()
-    {
-        //Get things going.  Request focus, set size, et cetera
-        setSize (200,200);
-        setVisible(true);
-        validate();
-
-        f = new Frame("The frame SHOULD be shown MAXIMIZED");
-        f.setSize(300, 300);
-        f.setLocation(50, 50);
-        f.setExtendedState(Frame.MAXIMIZED_BOTH);
-        f.setVisible(true);
-    }// start()
-
-}// class InitialMaximizedTest
-
-/* Place other classes related to the test after this line */
-
-
-
-
-
-/****************************************************
- Standard Test Machinery
- DO NOT modify anything below -- it's a standard
-  chunk of code whose purpose is to make user
-  interaction uniform, and thereby make it simpler
-  to read and understand someone else's test.
- ****************************************************/
-
-/**
- This is part of the standard test machinery.
- It creates a dialog (with the instructions), and is the interface
-  for sending text messages to the user.
- To print the instructions, send an array of strings to Sysout.createDialog
-  WithInstructions method.  Put one line of instructions per array entry.
- To display a message for the tester to see, simply call Sysout.println
-  with the string to be displayed.
- This mimics System.out.println but works within the test harness as well
-  as standalone.
- */
-
-class Sysout
-{
-    private static TestDialog dialog;
-
-    public static void createDialogWithInstructions( String[] instructions )
-    {
-        dialog = new TestDialog( new Frame(), "Instructions" );
-        dialog.printInstructions( instructions );
-        dialog.setVisible(true);
-        println( "Any messages for the tester will display here." );
+public class InitialMaximizedTest {
+
+    static Frame frame;
+
+    public static void main(String[] args) throws Exception {
+        if (!Toolkit.getDefaultToolkit()
+                    .isFrameStateSupported(Frame.MAXIMIZED_BOTH)) {
+            return;
+        }
+
+        Robot robot = new Robot();
+        try {
+            EventQueue.invokeAndWait(InitialMaximizedTest::createAndShowFrame);
+            robot.waitForIdle();
+            robot.delay(2000);
```

Comment:
delay is too long ? Generally we have 1000 ms delay after creating test UI.

Reply:
Reduced